src/ClientWidgetApi.ts

```diff
@@ -140,7 +168,9 @@ export class ClientWidgetApi extends EventEmitter {
         ).then(caps => {
             return this.driver.validateCapabilities(new Set(caps.capabilities));
         }).then(allowedCaps => {
+            console.log(`Widget ${this.widget.id} is allowed capabilities:`, Array.from(allowedCaps));
             this.allowedCapabilities = allowedCaps;
+            this.allowedEvents = Array.from(new Set(WidgetEventCapability.findEventCapabilities(allowedCaps)));
```

Comment:
Is there any reason for `findEventCapabilities` not to do the deduping?

Reply:
I don't actually know why there's deduping here.